source/extensions/network/dns_resolver/cares/dns_impl.h

```diff
@@ -107,6 +109,13 @@ class DnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logger::I
   private:
     void startResolutionImpl(int family);
 
+    struct AvailableInterfaces {
```

Comment:
nit: It would be nice to add a brief comment here. It's a pretty readable struct, but comments are always nice. (In particular, it would be good to clarify the meaning of "available" in this context)